lib/Dancer/Serializer/XML.pm

```diff
@@ -71,7 +83,18 @@ sub deserialize {
     }
 
     %options = (%options, @_);
+    # This is the promised terrible hack: claim that the LWP-talking code has
+    # already been loaded, and make sure that the handler that's called when
+    # we're dealing with an external entity does nothing.
+    # For whichever reason, this handler is called despite XML::Parser
+    # (which on my machine is the only XML::SAX backend that can handle
+    # external entities) having a ParseParamEnt option which is off by default,
+    # but appears to only be used deep in the XML::Parser XS guts.
+    no warnings 'redefine';
+    local *XML::Parser::lwp_ext_ent_handler = sub { return };
+    local $INC{'XML/Parser/LWPExternEnt.pl'} = 'Do not load this!';
```

Comment:
I'll tweak the value to be e.g. 'Dancer::Serializer::XML disabled loading this to patch around XXE vulnerabilities' or similar, just in case it saves anyone some head-scratching in future.